### PR TITLE
Add zipped mongodb and genenotebook datatypes

### PR DESCRIPTION
**Pull request by abretaud**
Adding two new datatypes:
- mongodb = a tar.bz2 archive of a mongodb database
- genenotebook = a mongodb archive, specialised for the output of the [genenotebook tool](genenotebook tool) and the corresponding [genenotebook gxit I'm working on](https://gitlab.com/ifb-elixirfr/usegalaxy-fr/galaxy/-/blob/release_23.0_ifb/tools/interactive/interactivetool_genenotebook.xml)

## How to test the changes?
There are some code examples in the comment, I'm not sure if it's used by automated tests or just comments.
I've installed it manually on a local server, it seems to work well

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/datatypes/binary.py [hunk 1]
```diff
@@ -3516,6 +3517,90 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
...
+    def sniff(self, filename: str) -> bool:
+        if filename and tarfile.is_tarfile(filename):
+            with tarfile.open(filename, "r") as temptar:
+                return "mongo_db/_mdb_catalog.wt" in temptar.getnames()
+        return False
+
+    def set_peek(self, dataset: DatasetProtocol, **kwd) -> None:
+        if not dataset.dataset.purged:
+            dataset.peek = f"MongoDB Archive ({nice_size(dataset.get_size())})"
+            dataset.blurb = "MongoDB version %s" % (dataset.metadata.version or "unknown")
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
            dataset.blurb = f'MongoDB version {dataset.metadata.version or "unknown"}'
```

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -3516,6 +3517,90 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
...
+                            vchunk = md_version_file.read(500)
+
+                            version_match = re.match(b".*version\x00\x06\x00\x00\x00([0-9.]+).*", vchunk)
+
+                            if version_match:
+                                dataset.metadata.version = util.unicodify(
+                                    version_match.group(1).decode("utf-8")
+                                ).strip()
+        except Exception as e:
+            log.warning("%s,(CompressedArchive set_meta Exception: %s", self, util.unicodify(e))
```
Comment (jdavcs) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
            log.warning(f"{self} CompressedArchive set_meta Exception: {util.unicodify(e)}"
```
Did I get this right?

File: lib/galaxy/datatypes/binary.py [hunk 3]
```diff
@@ -3516,6 +3517,90 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
...
+                            vchunk = md_version_file.read(500)
+
+                            version_match = re.match(b".*version\x00\x06\x00\x00\x00([0-9.]+).*", vchunk)
+
+                            if version_match:
+                                dataset.metadata.version = util.unicodify(
+                                    version_match.group(1).decode("utf-8")
+                                ).strip()
+        except Exception as e:
+            log.warning(f"{self} CompressedArchive set_meta Exception: {util.unicodify(e)}"
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/binary.py]:
```suggestion
            log.warning("%s CompressedArchive set_meta Exception: %s", self, e)
```

is the most efficient, but not a biggie.

**Review by jdavcs — APPROVED** [review]
Thank you for addressing the comments! Everything looks good to me.

**Comment by abretaud** [comment]
Thanks for the reviews/merge!

**Pull request merged**